cypress-io/cypress
# feat: Enable create from Vue component for projects with custom spec patterns

- Closes #23071

### User facing changelog


This is a part of the create from Vue component effort. It shouldn't have its own changelog entry

### Additional details


In the last create from Vue component PR, I excluded projects with custom spec patterns to separate out complexity. In order to release the feature, this also needs to work for projects with custom spec patterns.

### Steps to test


1. Open a Vue project that has a custom spec pattern defined for CT
2. If your project has specs, click on the "+ New Spec" button
3. Verify that you can see the "Create from component" card
4. Click on the card and then choose a Vue component from the list.
5. Verify that the spec file that is created matches your project's spec pattern and that the component you selected is imported correctly into the spec file (best way to verify this is to run the new spec and ensure that the spec mounts correctly).

### How has the user experience changed?


Users will now see the "Create from component" generator card when adding a new spec to a Vue project using component testing **when the project has a custom spec pattern** defined in its configuration.

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
